### ISSUE TITLE
Support pandas Float64Dtype in boxenplot

### ISSUE THREAD

**Issue by afrendeiro** [mwaskom/seaborn#2434]
Pandas 1.2.0 added a [Float64DType](https://pandas.pydata.org/pandas-docs/stable/whatsnew/v1.2.0.html?highlight=float64dtype#experimental-nullable-data-types-for-float-data) type extension. Seaborn boxenplot fails with that input:
```python
data = pd.DataFrame(
   {"cont": np.random.random(20), "cat": np.random.choice(["a", "b"], 20)}
).convert_dtypes()

sns.boxplot(data=data, x='cat', y='cont')  #  okay
sns.boxenplot(data=data, x='cat', y='cont')  # error
```
At the same time Barplot and Boxplot don't fail. This is because boxenplot is at some point converting the input into a numpy array which makes the array dtype 'object'. That dtype fails when is input to `np.isfinite` [here](https://github.com/mwaskom/seaborn/blob/d334ea6b50dd8e5101c7b4c71b500cf01350499b/seaborn/categorical.py#L1843).

**Comment by mwaskom** [comment]
This is something that should magically start to work once the work in https://github.com/mwaskom/seaborn/issues/2429 is complete.

**Comment by afrendeiro** [comment]
Cool, sounds good! Feel free to close the PR or just recycle the test case.

**Comment by mwaskom** [comment]
Closed by #2435
